packages/http-client-java/generator/http-client-generator-core/src/main/java/com/microsoft/typespec/http/client/generator/core/template/EnumTemplate.java

```diff
@@ -150,7 +150,10 @@ protected void writeBrandedExpandableEnum(EnumType enumType, JavaFile javaFile,
                 addGeneratedAnnotation(classBlock);
                 classBlock.annotation("Override");
                 classBlock.method(JavaVisibility.Public, null, "boolean equals(Object obj)",
-                    function -> function.methodReturn("Objects.equals(this.value, obj)"));
+                    function -> function.line(
+                        "if (this == obj) {\n" + "    return true;\n" + "}\n" + "if (!(obj instanceof %1$s)) {\n"
+                            + "    return false;\n" + "}\n" + "return Objects.equals(this.value, ((%1$s) obj).value);",
+                        enumName));
```

Comment:
Can we write it line by line? Current code is not easy to read.

Multiple `function.line` would do, though better via `if()`,

Reply:
Sure, let me break them into lines.